packages/core/src/integrations/mcp-server/resultExtraction.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Result extraction functions for MCP server instrumentation
+ *
+ * Handles extraction of attributes from tool and prompt execution results.
+ */
+
+import {
+  MCP_PROMPT_RESULT_DESCRIPTION_ATTRIBUTE,
+  MCP_PROMPT_RESULT_MESSAGE_COUNT_ATTRIBUTE,
+  MCP_TOOL_RESULT_CONTENT_COUNT_ATTRIBUTE,
+  MCP_TOOL_RESULT_IS_ERROR_ATTRIBUTE,
+} from './attributes';
+
+/**
+ * Build attributes for tool result content items
+ * @param content - Array of content items from tool result
+ * @returns Attributes extracted from each content item including type, text, mime type, URI, and resource info
+ */
+function buildAllContentItemAttributes(content: unknown[]): Record<string, string | number> {
+  const attributes: Record<string, string | number> = {
+    [MCP_TOOL_RESULT_CONTENT_COUNT_ATTRIBUTE]: content.length,
+  };
+
+  for (const [i, item] of content.entries()) {
+    if (typeof item !== 'object' || item === null) continue;
+
+    const contentItem = item as Record<string, unknown>;
```

Comment:
l: generally I prefer if we use functions to type narrow instead of casts, but feel free not to change.

```ts
function isValidContentItem(item: unknown): item is Record<string, unknown> {
  return item != null && typeof item === 'object';
}
```

```ts
if (!isValidContentItem(item)) {
  return;
}

// continue with rest of code.
```